src/client/vzReducer/searchReducer.ts

```diff
@@ -0,0 +1,68 @@
+import { VZAction, VZState } from '.';
+import { SearchFile, SearchFileVisibility, SearchResult, ShareDBDoc, VZCodeContent } from '../../types';
+
+function searchPattern(shareDBDoc: ShareDBDoc<VZCodeContent>, pattern: string): SearchResult {
+  const files = shareDBDoc.data.files
+  const fileIds = Object.keys(shareDBDoc.data.files);
+  let results: { [id: string] : SearchFile } = {};
+
+  for (let i = 0; i < fileIds.length; i++) {
+    const file = files[fileIds[i]];  
+
+    if (files[fileIds[i]].text) {
+      const fileName = file.name;
+      const lines = file.text.split("\n");
+      const matches = [];
+
+      for (let j = 0; j < lines.length; j++) {
+        const index = lines[j].indexOf(pattern);;
```

Comment:
Very nice implementation for search! I'm curious how fast this is, as it's a brute force algorithm. It's probably fine, but I'd be curious to make unit tests for this function and see how fast it is for various sizes of file trees.